man/machine-info.xml

```diff
@@ -128,6 +128,13 @@
         specific as <literal>Left Rack, 2nd Shelf</literal>.
         </para></listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>KERNEL_INSTALL_MACHINE_ID=</varname></term>
+
+        <listitem><para>Specifies the machine ID <command>kernel-install</command> should use. The value must
+        be a valid machine ID (32 hexadecimal characters).</para></listitem>
```

Comment:
How do you envision this working specifically? With the current approach, my idea was that initially, image builders (mkosi) would write KERNEL_INSTALL_MACHINE_ID to /etc/machine-info themselves to force kernel-install to use that ID instead of the machine ID generated by the systemd rpm post install scriptlet. Eventually we'd then hopefully fix distros to not generate machine IDs in post-install scripts of systemd but do so in Anaconda or equivalent after which we wouldn't need to do anything special in mkosi anymore since kernel-install would itself generate a new UUID and store it in machine-info.

Did you have the same idea? Or should kernel-install store the initial machine ID generated by the rpm post install scriptlet in KERNEL_INSTALL_MACHINE_ID in /etc/machine-info itself so that it doesn't matter if it is removed later on in the image build process?